### PR TITLE
Add support for updating HassOS and Hass core

### PR DESCRIPTION
**Pull request by mjkl-gh**
## Description
I've added 2 files for HassOS and Hass core to be updated by this ansible script

## Related issues

#6 

## Motivation and Context

Since i'm running HassOS on a virtual machine I would like to write a playbook for updating all of my servers and applications. This collection does part of what I want but was missing key features for this

## Types of changes

- [ ] Docs change / refactoring.
- [X] Non-breaking change (fix or feature that wouldn't cause existing functionality to change/break).
- [ ] Breaking change (fix or feature that would cause existing functionality to change/break).

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes don't generate new warnings.
- [x] I have read the [CONTRIBUTING](https://github.com/totaldebug/.github/blob/master/docs/CONTRIBUTING.md) document.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests pass.

### CODE REVIEW

File: plugins/modules/hassio_core.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      - State of host
+    required: true
+    choices: ['restarted', 'updated', 'stop']
+"""
+
+EXAMPLES = """
+# restart HassIO Core
+- hassio_host:
+    state: restarted
+"""
```
Comment (marksie1988) [inline plugins/modules/hassio_core.py]: documentation needs to be updated.

File: plugins/modules/hassio_os.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - State of host
+    required: true
+    choices: ['updated']
+"""
+
+EXAMPLES = """
+# Update HassIO OS
+- hassio_host:
+    state: update
+    token: <SUPERVISOR_TOKEN>
```
Comment (marksie1988) [inline plugins/modules/hassio_os.py]: documentation needs to be updated.

**Review by marksie1988 — CHANGES_REQUESTED** [review]
Please update the documentation for the commands, once done I can approve this and merge :)

File: plugins/modules/hassio_core.py (hunk 1 above)
Comment (mjkl-gh) [reply plugins/modules/hassio_core.py]: I've updated documenation can you check if you agree?

File: plugins/modules/hassio_os.py (hunk 2 above)
Comment (mjkl-gh) [reply plugins/modules/hassio_os.py]: I've updated documenation can you check if you agree?

**Review by marksie1988 — APPROVED** [review]

**Comment by mjkl-gh** [comment]
I fixed the formatting issues from black as well

**Comment by mjkl-gh** [comment]
@marksie1988  I've also added a facts module to get info for conditional plays, maybe you want to check that one out as well?

**Comment by mjkl-gh** [comment]
@marksie1988 Just a friendely reminder these changes should be ready to be merged? 

Or is there anything I should change/help with?

**Pull request merged**